sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -349,21 +146,26 @@ namespace Azure { namespace Core {
      * @return A value associated with the context found; an empty value if a specific value can't
      * be found.
      */
-    const ContextValue& operator[](const std::string& key) const
+    template <class T> const T& Get(const std::string& key) const
```

Comment:
@JeffreyRichter also suggests `GetWithDefaultValue`

Reply:
also we can still use `operator[]`, `operator[]` also supports template

Reply:
I'm open to suggestions for a clean operator[].  The only way I know to make it work is 
```C++
template <class T> const T& operator[](const std::string& key) const
```
Unfortunately the users of the context can't use it how you'd expect.
```C++
Context ctx;
auto& val = c3["key"]<int>();  //Invalid code

//This is how you must write it
//  Its much cleaner to use the Get method over this syntax
auto& val = ctx.operator[]<int>("key");
```

Reply:
```C++
auto& val = ctx["key"]<int>;
```

will that work?